Support EIP-4844 - WIP

dapplion opened the pull request:
**Motivation**

Full implementation of the EIP-4844 hard-fork logic. EIP-4844 rationale -> [B L O B S P A C E](https://twitter.com/protolambda/status/1584105020697432064)

**Description**

Implementation list:
- [x] Add params, config and types https://github.com/ChainSafe/lodestar/pull/4568
- [x] Make `c-kzg` library available for Lodestar. Add native bindings for `ethereum/c-kzg-eip4844` https://github.com/dankrad/c-kzg/pull/3
- [x] Block proposing
  - Add new engine API `engine_getBlobsV1`
  - Cache returned blobs in beacon to not modify validator cycle code
  - Publish to gossip topic, depending on fork
- [x] Block processing / import
  - Update chain `processBlock` to take block + (maybe) blobs pair
  - Validate blobs `is_data_available`, running KZG crypto
  - Store blobs to database + migrate from hot to cold on finalization + prune old blobs < MIN_EPOCHS_FOR_BLOBS_SIDECARS_REQUESTS
- [x] Add p2p ResResp methods `BeaconBlockAndBlobsSidecarByRoot` + `BlobsSidecarsByRange`
  - Update register / unregister logic to active those topics only post-eip4844
- [x] Add p2p gossip topic `beacon_block_and_blobs_sidecar`

Test list:
- [ ] Integrate with interop repo https://github.com/Inphi/eip4844-interop/pull/66

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ChainSafe/lodestar?pullRequest=4774) <br/>All committers have signed the CLA.

g11tech commented on packages/beacon-node/src/api/impl/beacon/blocks/index.ts:
```diff
@@ -186,17 +189,24 @@ export function getBeaconBlockApi({
+
+        () =>
+          chain.processBlock(blockForImport).catch((e) => {
+            if (e instanceof BlockError && e.type.code === BlockErrorCode.PARENT_UNKNOWN) {
```
what scenario this would be possible, PBS?

dapplion replied:
This is an API, an external actor can submit a block for unknown parent. In fact this is exactly what we do in the unknown root sync test from @nazarhussain

g11tech commented on packages/beacon-node/src/chain/blocks/importBlock.ts:
```diff
@@ -305,8 +305,18 @@ export async function importBlock(
   await this.db.block.add(block);
 
+  if (this.config.getForkSeq(block.message.slot) >= ForkSeq.eip4844) {
+    if (!blobs) {
```
should this extraction and error throw be before forkChoice.onBlock...

dapplion replied:
yes, our import block flow is unsafe because it can end up in an inconsistent state

g11tech commented on packages/beacon-node/src/chain/produceBlock/computeNewStateRoot.ts:
```diff
@@ -22,6 +27,9 @@ export function computeNewStateRoot(
     blockEmptySig,
+    // ExecutionPayloadStatus.valid: Assume payload valid, it has been produced by a trusted EL
+    // DataAvailableStatus.available: Assume the blobs to be available, have just been produced by trusted EL
+    {executionPayloadStatus: ExecutionPayloadStatus.valid, dataAvailableStatus: DataAvailableStatus.available},
```
we generally use `ExecutionStatus` from forkchoice outside the engine module

dapplion commented:
@nazarhussain sim tests failed with 
```
file:///home/runner/work/lodestar/lodestar/packages/utils/src/sleep.ts:24
      reject(new ErrorAborted());
             ^
ErrorAborted: Aborted 
    at AbortSignal.onAbort (file:///home/runner/work/lodestar/lodestar/packages/utils/src/sleep.ts:24:14)
    at AbortSignal.[nodejs.internal.kHybridDispatch] (node:internal/event_target:736:20)
    at AbortSignal.dispatchEvent (node:internal/event_target:678:26)
    at abortSignal (node:internal/abort_controller:292:10)
    at AbortController.abort (node:internal/abort_controller:323:5)
    at SimulationEnvironment.stop (file:///home/runner/work/lodestar/lodestar/packages/cli/test/utils/simulation/SimulationEnvironment.ts:210:29)
    at Timeout._onTimeout (file:///home/runner/work/lodestar/lodestar/packages/cli/test/utils/simulation/SimulationEnvironment.ts:140:20)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
```
after apparently completing successfully, can you look into it?

nazarhussain commented:
@dapplion This one failed because the nodes could not sync and tests run out of expected timeout. 

```
Simulation environment "multi-fork" is stopping: On timeout
```

dapplion commented:
Note: https://github.com/Inphi/eip4844-interop repo uses a modified minimal preset (SLOTS_PER_EPOCH = 3) which breaks a bunch a tests here

twoeths commented on packages/beacon-node/src/network/reqresp/handlers/blobsSidecarsByRange.ts:
```diff
@@ -0,0 +1,93 @@
+  // Finalized tram of blobs
+  // TODO EIP-4844: Should the finalized block be included here or below?
+
+  if (endSlot <= finalizedSlot) {
```
startSlot

dapplion committed:
Implement EIP-4844 WIP

commit f1e4aaa775da133b6a28ae819acf87d74a5fc485
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Dec 1 15:31:01 2022 +0700

    Fix get state for attestation verification logic

commit ad2c3009be9e96d7f4bafc0485e42798ad7d43ec
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Dec 1 15:14:14 2022 +0700

    Add eip4844 sim test

commit 6bf2e623b63398dd2e06e209cf8a1cfeb14f9fa8
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Dec 1 14:54:45 2022 +0700

    Fix typo in gossipsub coreTopicsAtFork

commit 2e565a82d55bc310230009386248dda7468aa70b
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Dec 1 14:54:08 2022 +0700

    Fix typo in updateSyncState currentEpoch

commit 4169498814b5d62ae3ede34ffa46e1da2fc4faac
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Dec 1 14:53:54 2022 +0700

    Create valid blobs in mock

commit 0f22a803440c955b6f14690b1bdf1e21932f96ed
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Dec 1 00:50:08 2022 +0700

    Debug failing sim tests with more better progress logs

commit 32b177939186e2b60b2d270eb05654132e12bde0
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 30 22:19:54 2022 +0700

    Don't create invalid mock data

commit d77e74106f8ecf8bb173cbc5d13bda38b41b7387
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 30 15:52:08 2022 +0700

    Allow to run mock EL in sim tests

commit 6fd5d465c2b8b31055a869e6371b55b9daae498e
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 30 15:51:25 2022 +0700

    Log chain config on start

commit fb58663f4cf12d9f97079343b2ef6cfc26f0edc2
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 30 15:50:36 2022 +0700

    Add disabledWithdrawals

commit ad9b478522b426e8c9e1030bd7bc5d18707cd4c8
Author: gajinder <develop@g11tech.io>
Date:   Wed Nov 30 13:25:52 2022 +0530

    add eip4844 interop

commit 74287e37e6a1b8041fb1b70891c9c960b4afde9e
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 30 00:22:36 2022 +0700

    Skip blobs validation if from gossip

commit b30aa71e34c1de83d06ed24e659a9f8e535f7fb0
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 30 00:09:16 2022 +0700

    Use enum for BlockImport

commit 4bf587dad52f84fc0da5cc3d90b8955b41e9ff35
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sun Nov 27 12:38:07 2022 +0700

    Add note on reorg blobs

commit ccdb9bb7b27fd35e9a8aa9f85be69812eeb1b009
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sun Nov 27 09:33:29 2022 +0700

    Prune producedBlobsSidecarCache

commit ebf0eae926d6f71c50fe7d95b02d83c19455205b
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sun Nov 27 09:28:05 2022 +0700

    Subscribe beacon_block_and_blobs_sidecar on fork

commit 8614743e0d8fa2cbb7218041d9af6105f3542115
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sun Nov 27 09:27:51 2022 +0700

    Simplify blobsSidecar cache type

commit 58c595a61da5f850681533ccc4418d3a2aa0dfeb
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sun Nov 27 02:00:47 2022 +0700

    Fix rebase issues

commit 804887df42f782fe72cd507d013b2462d85c0ba3
Merge: 22d1f85d5d c1ab090194
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sun Nov 27 01:56:37 2022 +0700

    Merge remote-tracking branch 'origin/unstable' into dapplion/eip-4844

commit 22d1f85d5d2df77ff91d215f7598948847aab5de
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 26 23:51:21 2022 +0700

    Fix types

commit 021609989eeed8b04ff7dbdfe22793fa04c7266b
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 26 23:40:57 2022 +0700

    fix rebase issues

commit 67962076b1a429d46f78b10f1056072334c49e16
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 26 23:40:28 2022 +0700

    Subscribe to reqresp methods on forks

commit 99cf400b79f9372441b7d4f59e38d93d7ccd5d47
Merge: d33472bed6 798ec16d30
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 26 21:35:52 2022 +0700

    Merge remote-tracking branch 'origin/unstable' into dapplion/eip-4844

commit d33472bed68f3787c8334de9488680ff64e0291a
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 26 11:05:06 2022 +0700

    Mock blobs WIP

commit 1dd5e74fe03145b08ddc28bc1b2012be07edd17f
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 25 00:02:41 2022 +0300

    Enable blobs reqresp

commit 9c0a0e67987f5d2ddffb09f35df96c1cf0ceb1b7
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 25 00:02:18 2022 +0300

    Add yamux

commit 0427435f0679febdaf32b93ed3d3f9c7a5230340
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 19 01:45:11 2022 +0100

    Temp change FIELD_ELEMENTS_PER_BLOB

commit 370c733e4a6621e8c4118f1bcb87a84453a96e05
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 19 01:44:30 2022 +0100

    Ensure correct sizes on blob bundle

commit 90f0d4c6c7cbddfb84f62d759dbd21150ba3455c
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Sat Nov 19 00:57:47 2022 +0100

    Fix verifyKzgCommitmentsAgainstTransactions math

commit 5a6652bbf1b4727a89551dfba074cc4098096eca
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 23:12:46 2022 +0100

    Check allVersionedHashes length

commit 37f33870cf3abeac3c81efa7d9b4cded34955555
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 22:40:36 2022 +0100

    Skip verifyAggregateKzgProof is empty blobs

commit e5d6f4fb58ed78b7e0dad5f53544ca7433163566
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 22:15:30 2022 +0100

    Set SLOTS_PER_EPOCH: 3 for interop tests

commit 8f18bd8aa590bce52b6fc87e3907f609561521cc
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 21:51:06 2022 +0100

    Move trusted setup file out of src

commit 8d0f1c0a67963912feb0314bb7d290eec67f2231
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 02:01:19 2022 +0100

    Revert fork-choice changes

commit 3b82bd70059fac1d632d0c07bd46ff651b805d11
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 01:57:44 2022 +0100

    Only line JSON

commit b47162b282d7e67699ca856d8abf1e18ebf0faad
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 01:49:29 2022 +0100

    Revert preset changes

commit c34e969f074368b22f6a31665aa70f4c9905f206
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 01:18:23 2022 +0100

    Fix test bellatrix/fork_choice/on_merge_block/pyspec_tests/all_valid

commit 9ea834885670bfb156a93d3f1c2bec61b7c23a72
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 00:43:16 2022 +0100

    Fix test Should subscribe to gossip core topics on demand

commit 58fb922861786f48111eb62c5df5cbb250010581
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Fri Nov 18 00:40:26 2022 +0100

    Fix test utils / params / assertEqualParams

commit 0d49fa4e8c0810a2a25f2820092914cb834066f8
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 23:33:37 2022 +0100

    Fix gossipsub test

commit 7ae69defd4c61b36498537df41ff3e2d199bd34f
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 23:29:23 2022 +0100

    Fix UnknownBlockSync test

commit 4e2a63a89935c400f7e72b9e6abc9796c448334d
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 23:29:12 2022 +0100

    Remove useless test

commit 4f544dffb9d7ea718e9816c93040602ca18825f7
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 23:23:25 2022 +0100

    Fix enrForkID.nextForkVersion test

commit a3f479e2aa1c5c70aeab3d6efd6b905264b4a8d6
Author: Lion - dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 19:09:15 2022 +0100

    Disable parallel for spec tests (#4776)

commit e7eba90c25f37ad822ad3a4c29585ef5fc5658f0
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 19:28:26 2022 +0100

    Fix assertLinearChainSegment

commit 864b7d33067c1a975f7bdcee0d960313c097a5ba
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 18:06:13 2022 +0100

    Review PR

commit 6963c1c549f4ebc535dfbafb38e515ffef9216c5
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 16:23:00 2022 +0100

    FIx readme checks

commit ece46f0faaf915f6756e3739c8d69cce1758f234
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 15:43:03 2022 +0100

    Fix tests

commit 6c878e5791e7ad3990191cf43c5bd2a189f53cdc
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 14:55:05 2022 +0100

    Fix topic test

commit 0f44e49e365612f1f1b8e3742e9f7d95332933c9
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 14:54:05 2022 +0100

    Move kzg test to beacon-node

commit 3f7a0ad641d02b1255805223ed9ef6c71f51f736
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 14:46:30 2022 +0100

    Fix types

commit 27e503fe00c8888ac2ff6703d55b480d78481d07
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 14:46:15 2022 +0100

    Remove old trusted setup

commit fdeb4f59a10fe9dfffad7e9655ce5fc21bf7d662
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 14:36:35 2022 +0100

    Review PR pt.2

commit 7da555ebe7a86f48a8cd226bd09d387e8596e010
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 12:10:04 2022 +0100

    Remove   DOMAIN_BLOBS_SIDECAR

commit ceea759fa94f644ae8b88244e61ee20a63605b1c
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 12:06:59 2022 +0100

    Ensure forks are not scheduled in presets

commit 414878516f575238194cba1ba461908fc76afdaa
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 12:06:09 2022 +0100

    Review Dan PR

commit e969457027ec6b927cccd0a472b98e6e28e1d78b
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 11:03:44 2022 +0100

    Fix types

commit e16a699d4ab35d58bc5d30d577dd4fb3fee0818c
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 09:53:17 2022 +0100

    Implement blobsSidecar pruning

commit c90f0fbf1c1f296db507c2d25a07594d43daf11c
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 09:42:05 2022 +0100

    Archive blobs

commit 4632bcfed54be1fa2c34bef61d1d3ffb521d73c4
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Thu Nov 17 09:41:54 2022 +0100

    Fix types

commit 46f58a4c7f5207b6a88f9e2869aa3183b23c1964
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 16 23:35:37 2022 +0100

    Clean up changes

commit 3d86a02ac0e25d0561a44258f4c0b540fe4a112d
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 16 21:00:33 2022 +0100

    WIP import block+blobs coupled

commit d2688155eb9849e588ee40f67da39eb095e132e4
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 16 18:57:22 2022 +0100

    WIP block processing

commit 175117d1dc834d9cfb23e7349219b1e883e3f1ed
Merge: ffb6336d20 bfe5da8b9e
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Wed Nov 16 11:44:02 2022 +0100

    Merge remote-tracking branch 'origin/unstable' into dapplion/eip-4844

commit ffb6336d2000ac166ce42009f0baba17221189ae
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Tue Nov 15 21:27:30 2022 +0100

    WIP

commit 611a609bdda210435e1a65ac00576e839458cf9c
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Tue Nov 15 15:36:42 2022 +0100

    Fix script name

commit 26a0794bb4bdf0470dd08ae667260687adcde100
Author: dapplion <35266934+dapplion@users.noreply.github.com>
Date:   Tue Nov 15 15:33:38 2022 +0100

    De-couple block production

commit 513b8731eb4735122972e86c2c13dbd21001fe92
Author: dancoffman <coffman@coinbase.com>
Date:   Wed Nov 9 10:30:54 2022 -0800

    Bump c-kzg to version that handles empty array of blobs -- remove application code special-case

commit a63b043f0eac452733aea9b00f54b6a2c1bd7ab9
Author: dancoffman <coffman@coinbase.com>
Date:   Tue Nov 8 13:00:57 2022 -0800

    Update version of c-kzg to one which does not crash for blobs arrays of length 0 and 1

commit 6a8d53660ebbffda039d0cd443612f449be1f944
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 23:33:02 2022 -0800

    Delete unused polynomialCommitments file

commit 656b7d3a9aa5b421de206af2ddb74b4e9d26d183
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 23:26:44 2022 -0800

    processBlocks for 4844 blocks. Do not compute aggregate proof for empty array (it crashes)

commit aee7d9512879ea9d7b0b91827fccf98a4b20ee58
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 16:53:02 2022 -0800

    Bump c-kzg again

commit b992fc69d545c3079e9afa202cd0b0dd17f47fe7
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 16:52:09 2022 -0800

    Add unit test just to verify the c-kzg library works

commit 054cfb6822901fffa2e49f57c1633980453b45a1
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 15:17:50 2022 -0800

    Fix trusted_setup file path

commit 7e8f3ab0474ebeb366a6f82491c2cb2f2536e116
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 11:55:34 2022 -0800

    Load KZG setup

commit 513e2772095c43d663b19ef4df5b71339e0de965
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 11:51:04 2022 -0800

    Bump c-kzg to the version that compiles on Linux

commit 78220efb9213fceb2ff45a665be26d5eda34439f
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Nov 7 11:32:30 2022 -0800

    Bump @types/node

commit 3db4f5770ed42c932e8a19599165140ac084fa80
Author: dancoffman <coffman@coinbase.com>
Date:   Sat Nov 5 01:17:08 2022 -0700

    Use computeAggregateKzgProof from the c-kzg package and delete all the code it encapsulates

commit 2fe076b052c149d46abfb3968a714aebe4ca123f
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Oct 31 17:36:49 2022 -0700

    Drill blobs all the way through

commit ad1a62fab0c2067ab3744d41ce30139170020f94
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Oct 31 16:10:33 2022 -0700

    Use SSZ object correctly

commit 28e9f8a1421158ffcd2657c654dba0d152c61d93
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Oct 31 15:11:59 2022 -0700

    publishBlockWithBlobs implementation. A lot more moon math scaffolding that will be removed.

commit d1e3e4b59dcf4e409d488b8b2bed30d3fdb5fcaa
Author: dancoffman <coffman@coinbase.com>
Date:   Sat Oct 29 00:23:42 2022 -0700

    Scaffold block proposing w/ KZG commitments

commit f3cedf932946530787a8fc92c147708a4fbca45e
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 28 22:28:22 2022 -0700

    Add a couple more of the polynomial commitment types

commit 2e9e41a5ac1e70fb32ecdfbd95cb9c3bfeea1299
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 28 16:32:14 2022 -0700

    Add chain config constants and validator types

commit b1663874d4a5e461a53c958b034cbc31d06eb73b
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 28 15:28:25 2022 -0700

    Fix assigning excessDataGas to ExecutionPayload

commit 2a4bedb738571869cc045e671ecca7946621c03e
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 28 12:09:21 2022 -0700

    Revert Capella ssz type change

commit 89c43bf644d8e3832d250ce2d2ae6b83ca783edf
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 28 12:08:00 2022 -0700

    Revert upgradeStateToCapella

commit 87ecaafc13aa1baa7609765d1244a5c5d1e705c8
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 28 11:57:55 2022 -0700

    Fix allForks SSZ types

commit e6c13e6ba2a91d61173767f3dd581006dffbdb4a
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 28 12:13:17 2022 -0700

    Add beacon_block_and_blobs_sidecar gossip type

commit f5c8cfab0bb5f8658138c012038732e68c35500a
Author: dancoffman <coffman@coinbase.com>
Date:   Fri Oct 21 11:54:04 2022 -0700

    Fix: Accidentally removed assigning execution payload to block

commit 78996a9f988afdf2d9ed88876b0ffe2deb3a7139
Author: dancoffman <coffman@coinbase.com>
Date:   Thu Oct 20 15:56:08 2022 -0700

    Fix state upgrade and patch blocks with missing fields when necessary

commit a83df9e54d97b2b02258c713279898f5a9046812
Author: dancoffman <coffman@coinbase.com>
Date:   Thu Oct 20 09:42:02 2022 -0700

    When producing a block to propose, assign blobKzgCommitments from Engine API getBlobsBundle response

commit 6a7630fbf873506b6253e3c20730e07e17657d44
Author: dancoffman <coffman@coinbase.com>
Date:   Wed Oct 19 16:33:41 2022 -0700

    Add engine_getBlobsBundleV1 Engine API call

commit 9de5e10b8e30d32f9a9869605a5097d33f8e8356
Author: dancoffman <coffman@coinbase.com>
Date:   Wed Oct 19 15:11:08 2022 -0700

    Capella and 4844 state upgrades

commit 71e7e0b36b41bc5e1c7a6bc2a0e9e2547199c8ce
Author: dancoffman <coffman@coinbase.com>
Date:   Wed Oct 19 12:08:06 2022 -0700

    Add DOMAIN_BLOBS_SIDECAR

commit d3fe373289af1007326a7a41a069cc51d047f131
Author: dancoffman <coffman@coinbase.com>
Date:   Wed Oct 19 12:05:04 2022 -0700

    Beacon chain types complete

commit 451f6e96e958e37a3f0464ca791be8348e896187
Author: dancoffman <coffman@coinbase.com>
Date:   Wed Oct 19 11:21:07 2022 -0700

    Spec-matching configs and presets

commit 266ba7a50bd806e45cff21d73b7735b8a6c9307c
Author: dancoffman <coffman@coinbase.com>
Date:   Tue Oct 18 18:56:33 2022 -0700

    Sketch out EIP-4844 types

commit 002d85090b4a5e1e069a3cb07bb87df4132d02ba
Author: dancoffman <coffman@coinbase.com>
Date:   Mon Oct 17 15:21:10 2022 -0700

    Updates to allow test:spec-min|main work, some edits to contributing instructions

fix build

Remove duplicate mentions of beacon_block_and_blobs_sidecar

Remove duplicate beaconBlockAndBlobsSidecarByRoot

Remove unwanted diff

run with withdrawals include

Fix sim tests types

Update command

update package

fix port

update the run

merge interop image update

move to time

Remove unwanted diff

Fix merge issue with sim tests

Fix code issue during merge conflicts

Remove a duplicate function decleration

remove bls-to-execution-change empty assignment

lint

fix issues

dapplion commented on packages/beacon-node/src/network/nodejs/bundle.ts:
```diff
@@ -35,6 +36,9 @@ export async function createNodejsLibp2p(options: ILibp2pOptions): Promise<Libp2
     }
   }
+
+  const yamuxInstance = yamux()();
```
Note to self: do not merge this change, yamux must be tested and merged on its own

g11tech committed:
do not request blobs if no blocks in range

g11tech committed:
increase req/resp timeout for testing

g11tech committed:
reduce diff

g11tech committed:
remove diff

g11tech committed:
reduce diff

dapplion commented:
Closing, base eip4844 features already merged.
